Lib/test/test_shutil.py

```diff
@@ -1666,12 +1666,18 @@ def check_chown(path, uid=None, gid=None):
         shutil.chown(dirname, group=gid)
         check_chown(dirname, gid=gid)
 
-        user = pwd.getpwuid(uid)[0]
-        group = grp.getgrgid(gid)[0]
-        shutil.chown(filename, user, group)
-        check_chown(filename, uid, gid)
-        shutil.chown(dirname, user, group)
-        check_chown(dirname, uid, gid)
+        try:
+            user = pwd.getpwuid(uid)[0]
+            group = grp.getgrgid(gid)[0]
+        except KeyError:
+            # In case the uid/gid cannot be resolved.
+            user = None
+            group = None
+        if user is not None and group is not None:
```

Comment:
You could do try/except KeyError: pass/else here instead of doing the assignments and None checks.

Reply:
Just wanted to keep the try/except scope as small as possible in case some of the other commands raise. Anyway will change.

Reply:
Oops, just realized what you meant here, re-submitted.